### PR TITLE
CUR2-3840: Add Uniswap V4 on X Layer to dex.trades

### PR DESCRIPTION
**Pull request by tomfutago**
## Thank you for contributing to Spellbook 🪄
Please open the PR in **draft** and mark as ready when you want to request a review.

## What & why
Add Uniswap V4 on X Layer to `dex.trades` so those swaps show up in the sector table.
This re-lands https://github.com/duneanalytics/spellbook/pull/9950 after the revert, without editing `uniswap_v4_chains()`. That one-line chain-list change was what forced a full refresh of every V4 chain (including `dex_bnb_base_trades`).
Known gap until a later chain-list PR: xlayer BaseAggregatorHook swaps stay in venue `dex.trades`, and `uniswap_v4_xlayer_aggregator_base_trades` is not in the cross-chain aggregator union.

## Architecture
- `dbt_subprojects/dex/models/trades/xlayer/platforms/uniswap_v4_xlayer_base_trades.sql` - venue V4 rows that feed `dex_xlayer_base_trades`
- `dbt_subprojects/dex/models/trades/xlayer/dex_xlayer_base_trades.sql` - chain union now includes V4
- +6 files: xlayer swaps/aggregator models, schema YAML, sources, seed rows
- Tested: original #9950 CI was green with 4 verified seed trades. Not tested: aggregator-hook classification on xlayer (hooks registry still excludes xlayer)
- Needs human eyes: CI `list downstream models` count (expect xlayer + `dex.trades` descendants, not the V4 fleet)
- Blast radius: `dex.trades` / `dex_xlayer.*` consumers. Deliberately not touching `uniswap_v4_aggregator_hooks` or `uniswap_v4_chains()`, which previously selected 300 downstream models

<details><summary>Agent context</summary>

- Re-land of #9950 minus `uniswap_v4_chains.sql`. Revert was #9953 after prod deploy selected `state:modified.macros+` off the shared chain list.
- `uniswap_v4_xlayer_swaps` still `ref('uniswap_v4_aggregator_hooks')`; that is a new model pointing at an unchanged hub, so hooks itself is not `state:modified`.
- Pool manager `0x360e68faccca8ca495c1b759fd9eee466db9fb32`, start date `2025-12-08`.
- Repro: `uv run dbt compile --select uniswap_v4_xlayer_swaps uniswap_v4_xlayer_base_trades dex_xlayer_base_trades --project-dir dbt_subprojects/dex/`
- Prior CI: https://github.com/duneanalytics/spellbook/actions/runs/32274109941
- Coverage ledger: no agent review yet on this re-land
</details>

Closes CUR2-3840
https://linear.app/dune/issue/CUR2-3840/add-x-layers-uniswap-v4-instance-to-dextrades

---
quick links for more information:
- [README.md](https://github.com/duneanalytics/spellbook/blob/main/README.md)
- [spellbook docs](https://github.com/duneanalytics/spellbook/tree/main/docs)
- [CONTRIBUTING.md](https://github.com/duneanalytics/spellbook/blob/main/CONTRIBUTING.md)

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: dbt_subprojects/dex/models/_projects/uniswap/xlayer/_schema.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      - dbt_utils.unique_combination_of_columns:
+          arguments:
+            combination_of_columns:
+              - tx_hash
+              - evt_index
+      - check_dex_aggregator_seed:
+          arguments:
+            blockchain: xlayer
+            project: uniswap
+            version: 4
```
[the remark this answers is not in the document]
Comment (tomfutago) [reply dbt_subprojects/dex/models/_projects/uniswap/xlayer/_schema.yml:55]: Known leftover from #9950, and expected on this re-land.

`uniswap_v4_chains()` is intentionally untouched, so xlayer is not in `uniswap_v4_aggregator_hooks`. Aggregator-hook swaps will not classify until a follow-up chain-list PR. Adding `dex_aggregator_seed` rows now would test a path that is empty by design.

Non-blocking for putting V4 into `dex.trades`.

**Comment by tomfutago** [comment]
CI is green. `list downstream models` on dex (deploy preview, `state:modified.body+` / `state:modified.macros+`):

- **71 models** (was **300** on #9950)
- **6 incremental** (was **177**): `uniswap_v4_xlayer.swaps`, `uniswap_v4_xlayer.aggregator_base_trades`, `dex_xlayer.base_trades`, `dex_xlayer.trades`, `dex_xlayer.token_volumes_daily`, `dex.trades`
- Remaining 65 are views (mostly project-level `*.trades` over `dex.trades`)
- No other-chain `dex_<chain>.base_trades` (no `dex_bnb_base_trades`)

Logs: https://github.com/duneanalytics/spellbook/actions/runs/32373672326

File: dbt_subprojects/dex/models/_projects/uniswap/xlayer/_schema.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomfutago) [reply dbt_subprojects/dex/models/_projects/uniswap/xlayer/_schema.yml:55]: Follow-up after checking on-chain: there are no samples to add that this model can emit.

`uniswap_v4_xlayer_aggregator_base_trades` is empty in CI (0 rows in `dune.dune_spellbook_ci__tmp_pr9954_32373672326_1.uniswap_v4_xlayer_aggregator_base_trades`). On-chain, no xlayer `creation_traces` since 2025-12-01 match the BaseAggregatorHook bytecode (`PUSH4 pollTokenJar` + `PUSH4 pseudoTotalValueLocked`), including all 192 non-zero hooks from `PoolManager_evt_Initialize`.

Adding `dex_aggregator_seed` rows would make this test fail (seed keys with no model match). Leave the test wired as a no-op until a BaseAggregatorHook actually lands on xlayer.

**Review by jeff-dude — APPROVED** [review]
